src/tests/conftest.py

```diff
@@ -20,30 +20,49 @@
 
 @pytest.fixture
 def check_duplicate_message_factory():
-    def _generate_error_message(cmd, args, aliases=None):
+    def _generate_error_message(cmd, args, aliases=None, delete_present=False):
         error_message = ""
         duplicates = set([arg for arg in args if args.count(arg) != 1])
 
         if duplicates:
             error_message += (
                 f"Duplicate arguments in %{cmd}. "
                 "Please use only one of each of the following: "
-                f"{', '.join(sorted(duplicates))}."
+                f"{', '.join(sorted(duplicates))}. "
             )
-            if aliases:
-                error_message += " "
 
-        if aliases:
+        if aliases and not delete_present:
             alias_list = []
             for pair in sorted(aliases):
                 print(pair[0], pair[1])
                 alias_list.append(f"{f'-{pair[0]}'} or {f'--{pair[1]}'}")
             error_message += (
                 f"Duplicate aliases for arguments in %{cmd}. "
                 "Please use either one of "
-                f"{', '.join(alias_list)}."
+                f"{', '.join(alias_list)}. "
             )
 
+        if delete_present:
+            delete_arg_list = [
+                "-d",
+                "-D",
+                "-A",
+                "--delete",
+                "--delete-force",
+                "--delete-force-all",
+            ]
+            delete_args_present = set()
+            for arg in args:
```

Comment:
this code seems repeated from the util function. I didn't understand why we need it here

Reply:
This function is a message generation factory so it generates the error message based on a set of inputs. As the edge case we're handling here is very specific, the logic for message generation is almost the same as that from the util function.

Reply:
Cant we just mention all the error messages in `parametrize` as in the other tests? There will be multiple cases but atleast the readability and debugging improves.

Reply:
Also, the error message suffix/ prefix can be mentioned as string constants and reused in the tests. so that if the message changes the tests also reflect the same.